packages/react-ai/src/components/AIConversation/context/ResponseComponentsContext.tsx

```diff
@@ -15,7 +28,9 @@ export const ResponseComponentsProvider = ({
   responseComponents?: ResponseComponents;
 }): JSX.Element => {
   return (
-    <ResponseComponentsContext.Provider value={responseComponents}>
+    <ResponseComponentsContext.Provider
+      value={prependResponseComponents(responseComponents)}
```

Comment:
This will create a new reference on each render, regardless of whether `responseComments` changes. Tbh kind of surprised that the React linting rules aren't complaining

Reply:
Yeah I guess I'm used to our linting rules complaining abou this if you send in an object like `{foo, bar}` so when this didn't complain i was a little unsure lol